Bibliotecas/Gym/Gym.ipynb

```diff
@@ -0,0 +1,294 @@
+    "env.close()                             # Fecha o ambiente"
+   ],
+   "outputs": [],
+   "metadata": {}
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "Se tudo correu bem, você deve ter visto algo como isso na sua tela:\n",
+    "\n",
+    "<center><img src=\"\" width=\"400\"/></center>\n"
+   ],
+   "metadata": {}
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "## Observações\n",
+    "Porém, como você deve ter notado, mexer aleatóriamente pra esquerda e pra direita não é bem um **aprendizado**, então é de se esperar que eventualmente seria interessante ensinar alguma coisa pro agente, porém como fazer isso?\n",
+    "\n",
+    "Primeiro, vale explicar o retorno da função `env.step()`, na verdade, ela retorna quatro valores que possibilitam implementar algoritmos de aprendizado por reforço, estes sendo:\n",
+    "<center>\n",
+    "\n",
+    "|Nome|Tipo|Descrição|\n",
+    "|-|-|-|\n",
+    "|`observation`|objeto|Um objeto especifico por ambiente que representa a observação do ambiente.|\n",
+    "|`reward`|float|Quantidade de recompensa alcançada pela última ação. A escala varia com o ambiente|\n",
+    "|`done`|boolean|Flag que indica se é hora de chamar o `reset()`. Indica que o episódio terminou.|\n",
+    "|`info`|dict|Informações diagnósticas úteis para debugar. Geralmente é bom pra estudar,<br> mas o seu agente não usa isso pra aprender|\n",
+    "\n",
+    "</center>\n",
+    "\n",
+    "Perceba que trata-se de uma implementação do ciclo de agente-ambiente, onde a cada timestep o agente scolhe uma açãõ e o ambiente retorna uma observação e uma recompensa.\n",
+    "\n",
+    "<center><img src=\"https://gym.openai.com/assets/docs/aeloop-138c89d44114492fd02822303e6b4b07213010bb14ca5856d2d49d6b62d88e53.svg\" width=\"300\"/></center>"
+   ],
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "source": [
+    "for i_episode in range(20):                                         # Para 20 episodios\n",
+    "    observation = env.reset()                   \n",
+    "    for t in range(100):                                            \n",
+    "        env.render()\n",
+    "        print(observation)                                          # Mostra as observações\n",
+    "        action = env.action_space.sample()\n",
+    "        observation, reward, done, info = env.step(action)\n",
+    "        if done:\n",
+    "            print(\"Fim do episdio em {} timesteps\".format(t+1))\n",
+    "            break\n",
+    "env.close()"
+   ],
+   "outputs": [],
+   "metadata": {}
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "A saída por enquanto é apenas um tanto de texto e números, note também que a mensagem que seria mostrada com o Done nunca é mostrada. Isso acontece por causa do ambiente escolhido, onde o done só acontece quando o carrinho sai da tela, o que muito provavelmente não aconteceu."
```

Comment:
O ```done``` também é ```True``` quando o episódio passa de 200 timesteps, mas nunca está chegando porque o for está de 0 a 199